mmv1/third_party/terraform/tests/resource_eventarc_channel_test.go.erb

```diff
@@ -0,0 +1,175 @@
+<% autogen_exception -%>
+package google
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"testing"
+
+	dcl "github.com/GoogleCloudPlatform/declarative-resource-client-library/dcl"
+	<% if version == "ga" -%>
+	eventarc "github.com/GoogleCloudPlatform/declarative-resource-client-library/services/google/eventarc"
+	<% else -%>
+	eventarc "github.com/GoogleCloudPlatform/declarative-resource-client-library/services/google/eventarc/beta"
+	<% end %>
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+)
+
+func TestAccEventarcChannel_BasicHandWritten(t *testing.T) {
+	t.Parallel()
+
+	region := getTestRegionFromEnv()
+	key1 := BootstrapKMSKeyWithPurposeInLocationAndName(t, "ENCRYPT_DECRYPT", region, "tf-bootstrap-key1")
+	key2 := BootstrapKMSKeyWithPurposeInLocationAndName(t, "ENCRYPT_DECRYPT", region, "tf-bootstrap-key2")
+
+	context := map[string]interface{}{
+		"billing_acct":  getTestBillingAccountFromEnv(t),
+		"org_id":        getTestOrgFromEnv(t),
+		"region":        region,
+		"project_name":  getTestProjectFromEnv(),
+		"key_ring":      GetResourceNameFromSelfLink(key1.KeyRing.Name),
+		"key1":          GetResourceNameFromSelfLink(key1.CryptoKey.Name),
+		"key2":          GetResourceNameFromSelfLink(key2.CryptoKey.Name),
+		"random_suffix": randString(t, 10),
+	}
+
+	vcrTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckEventarcChannelDestroyProducer(t),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccEventarcChannel_BasicHandWritten(context),
+			},
+			{
+				ResourceName:      "google_eventarc_channel.primary",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+			{
+				Config: testAccEventarcChannel_BasicHandWrittenUpdate0(context),
+			},
+			{
+				ResourceName:      "google_eventarc_channel.primary",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func testAccEventarcChannel_BasicHandWritten(context map[string]interface{}) string {
+	return Nprintf(`
+data "google_project" "test_project" {
+	project_id  = "%{project_name}"
+}
+
+data "google_kms_key_ring" "test_key_ring" {
+	name     = "%{key_ring}"
+	location = "us-central1"
+}
+
+data "google_kms_crypto_key" "key1" {
+	name     = "%{key1}"
+	key_ring = data.google_kms_key_ring.test_key_ring.id
+}
+
+  
+resource "google_kms_crypto_key_iam_binding" "key1_binding" {
+	crypto_key_id = data.google_kms_crypto_key.key1.id
+	role      = "roles/cloudkms.cryptoKeyEncrypterDecrypter"
+
+	members = [
+	"serviceAccount:service-${data.google_project.test_project.number}@gcp-sa-eventarc.iam.gserviceaccount.com",
+	]
+}
+
+resource "google_eventarc_channel" "primary" {
```

Comment:
nit: I think the spacing's off in this block, the contents should have one level of indent

Reply:
Acknowledged

Reply:
Done

Reply:
Done